Source/Scene/Scene.js

```diff
@@ -1161,28 +1196,126 @@ define([
+        for (var k = 0; k < scene._debugGlobeDepthCommands.length; ++k) {
+            destroyDebugGlobeDepthCommand(scene, k);
+        }
+        scene._debugGlobeDepthCommands.length = 0;
+    }
+
+    function updateDebugGlobeDepth(scene, context, uniformState, index) {
+        var texture = scene._debugGlobeDepthTextures[index];
+        var framebuffer = scene._debugGlobeDepthFramebuffers[index];
+        var command = scene._debugGlobeDepthCommands[index];
+
+        var width = context.drawingBufferWidth;
+        var height = context.drawingBufferHeight;
+
+        var textureChanged = !defined(texture) || texture.width !== width || texture.height !== height;
+        if (textureChanged) {
+            destroyDebugGlobeDepthTexture(scene, index);
+            destroyDebugGlobeDepthFramebuffer(scene, index);
+            destroyDebugGlobeDepthCommand(scene, index);
+
+            texture = context.createTexture2D({
+                width : width,
+                height : height,
+                pixelFormat : PixelFormat.RGBA,
+                pixelDatatype : PixelDatatype.FLOAT
+            });
+
+            scene._debugGlobeDepthTextures[index] = texture;
+
+            framebuffer = context.createFramebuffer({
+                colorTextures : [texture],
+                destroyAttachments : false
+            });
+
+            scene._debugGlobeDepthFramebuffers[index] = framebuffer;
+
+            command = context.createViewportQuadCommand(PassThrough, {
+                renderState : context.createRenderState(),
+                uniformMap : {
+                    u_texture : function() {
+                        return uniformState.globeDepthTexture;
+                    }
+                },
+                owner : scene
+            });
+
+            scene._debugGlobeDepthCommands[index] = command;
+        }
+    }
+
+    function executeDebugGlobeDepth(scene, context, passState, index) {
+        var fs =
+            'uniform sampler2D u_texture;\n' +
+            'varying vec2 v_textureCoordinates;\n' +
+            'void main()\n' +
+            '{\n' +
+            '    float z_window = texture2D(u_texture, v_textureCoordinates).r;\n' +
+//          '    gl_FragColor = vec4(mix(vec3(0.0), vec3(1.0), z_window), 1.0);\n' +
+            '    float n_range = czm_depthRange.near;\n' +
+            '    float f_range = czm_depthRange.far;\n' +
+            '    float z_ndc = (2.0 * z_window - n_range - f_range) / (f_range - n_range);\n' +
+            '    gl_FragColor = vec4(mix(vec3(0.0), vec3(1.0), z_ndc * 0.5 + 0.5), 1.0);\n' +
+//            '    float n = czm_currentFrustum.x;\n' +
+//            '    float f = czm_currentFrustum.y;\n' +
+//            '    float z_eye = (2.0 * n * f) / ((f - n) * z_ndc - f - n);\n' +
+//            '    gl_FragColor = vec4(mix(vec3(0.0), vec3(1.0), (-z_eye - n) / (f - n)), 1.0);\n' +
+            '}\n';
+
+        var c = context.createViewportQuadCommand(fs, {
```

Comment:
You don't need to create this command every frame that `debugShowGlobeDepth` is `true`. You can do something like:
```javascript
if (!defined(scene._depthViewportCommand)) {
    var fs = /* ... */
    scene._depthViewportCommand = context.createViewportQuadCommand(fs, {
        uniformMap : {
            u_texture : function() {
                return scene._currentDepthTexture;
            }
        },
        owner : scene
    });
}

scene._currentDepthTexture = scene._debugGlobeDepthTextures[index];
scene._depthViewportCommand.execute(context, passState);
```